maven/bnd-generate-maven-plugin/src/main/java/aQute/bnd/maven/generate/plugin/GenerateMojo.java

```diff
@@ -166,6 +167,10 @@ private GenerateOperation getOperation() {
 				}
 			} finally {
 				errors += BndContainer.report(project);
+				if (project.getTargetDir()
+					.exists()) {
+					IO.delete(project.getTargetDir());
+				}
```

Comment:
This seems wrong, like you are writing into a source folder and then need to tidy up after. It would be much better to write into an output folder originally. Can this be done?

Reply:
Yes and no. In Maven code is usually generated somewhere in the target directory and many people follow the mantra of not checking in generated code. With my main usecase being EMF, we have to abilitly to overwrite e.g. specific methods in the generated code and prevent them from being overwritten in the next run. Thus my output directory is my src/main/java folder.
Long story short, I achieve this by setting the directory of the bnd project to the directory of the maven Module, so everything is relative to this. Do you have any other idea on how to make something like this work without setting the workingDir to the module dir?
```
-generate: \
    src/main/resource/**.template;\
        generate="somegenerator";\
	output=src/main/java
```

The working dir for the Project is currently set here: https://github.com/bndtools/bnd/blob/355f1985af3ecf5c60de2458657a12298b3a9886/maven/bnd-generate-maven-plugin/src/main/java/aQute/bnd/maven/generate/plugin/BndContainer.java#L183

Reply:
If the user configures the generator to output in a source folder, that is their prerogative. But then you should not delete the folder since they may want to commit the changes.

Reply:
No, that is not the folder I delete. When the project is initialized, it automatically creates the `generate` dir, we usually have in the bnd workspace layout. I just realized that I can configure the projects target dir as well, so I pointed it to mavens target dir.